### PR TITLE
Add extra fields

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: data/transaction.go [hunk 1]
```diff
@@ -34,8 +32,11 @@ type Transaction struct {
 	ReceiverUserName     []byte        `json:"receiverUserName,omitempty"`
 	Logs                 *TxLog        `json:"logs,omitempty"`
 	HasSCR               bool          `json:"hasScResults,omitempty"`
+	SCCall               bool          `json:"isScCall,omitempty"`
```
Comment (bogdan-rosianu) [inline data/transaction.go:35]: `IsScCall` *

File: process/block/blockProcessor.go [hunk 2]
```diff
@@ -80,14 +81,49 @@ func (bp *blockProcessor) PrepareBlockForDB(
 		StateRootHash:         hex.EncodeToString(header.GetRootHash()),
 		PrevHash:              hex.EncodeToString(header.GetPrevHash()),
 		SearchOrder:           computeBlockSearchOrder(header),
-		AccumulatedFees:       header.GetAccumulatedFees().String(),
-		DeveloperFees:         header.GetDeveloperFees().String(),
 		EpochStartBlock:       header.IsStartOfEpochBlock(),
 	}
 
+	if header.GetAccumulatedFees() != nil {
```
Comment (iulianpascalau) [inline process/block/blockProcessor.go:87]: please extract header.GetAccumulatedFees() in a variable as to not call the method twice. Valid also for header.GetDeveloperFees()

File: process/block/serialize_test.go [hunk 3]
```diff
@@ -55,3 +56,29 @@ func TestBlockProcessor_SerializeEpochInfoData(t *testing.T) {
...
+			TotalNewlyMinted:                 "20",
+			RewardsPerBlock:                  "15",
+			RewardsForProtocolSustainability: "2",
+			NodePrice:                        "10",
+			PrevEpochStartRound:              222,
+			PrevEpochStartHash:               "7072657645706f6368",
+		},
+	})
+	require.Nil(t, err)
+	require.Equal(t, `{"nonce":0,"round":0,"epoch":0,"miniBlocksHashes":null,"notarizedBlocksHashes":null,"proposer":0,"validators":null,"pubKeyBitmap":"","size":345,"sizeTxs":0,"timestamp":0,"stateRootHash":"","prevHash":"","shardId":4294967295,"txCount":0,"accumulatedFees":"","developerFees":"","epochStartBlock":true,"searchOrder":1010,"epochStartInfo":{"totalSupply":"100","totalToDistribute":"55","totalNewlyMinted":"20","rewardsPerBlock":"15","rewardsForProtocolSustainability":"2","nodePrice":"10","prevEpochStartRound":222,"prevEpochStartHash":"7072657645706f6368"}}`, buff.String())
```
Comment (bogdan-rosianu) [inline process/block/serialize_test.go:83]: can you please add values in developer fees and other fields?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: process/block/blockProcessor.go (hunk 2 above)
Comment (miiu96) [reply process/block/blockProcessor.go:87]: done.

File: data/transaction.go (hunk 1 above)
Comment (miiu96) [reply data/transaction.go:35]: done./

File: process/block/serialize_test.go (hunk 3 above)
Comment (miiu96) [reply process/block/serialize_test.go:83]: done.

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

File: client/elasticClientCommon.go [hunk 4]
```diff
@@ -102,32 +102,68 @@ func elasticBulkRequestResponseHandler(res *esapi.Response) error {
...
+
+func extractErrorFromBulkBodyResponseBytes(bodyBytes []byte, index string) error {
+	if index == indexer.TransactionsIndex {
+		return searchForErrorsUpdate(bodyBytes)
+	}
+
+	return searchForErrorsIndex(bodyBytes)
+}
+
+func searchForErrorsUpdate(bodyBytes []byte) error {
```
Comment (bogdan-rosianu) [inline client/elasticClientCommon.go:116]: `searchForUpdateErrors`

File: client/elasticClientCommon.go [hunk 5]
```diff
@@ -102,32 +102,68 @@ func elasticBulkRequestResponseHandler(res *esapi.Response) error {
...
+			break
+		}
 	}
 
-	return nil
+	return fmt.Errorf("%s", errorsString)
 }
 
-func extractErrorFromBulkResponse(response *BulkRequestResponse) error {
+func searchForErrorsIndex(bodyBytes []byte) error {
```
Comment (bogdan-rosianu) [inline client/elasticClientCommon.go:146]: `searchForIndexErrors`

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: client/data.go [hunk 6]
```diff
@@ -9,16 +9,27 @@ const (
...
 	Errors bool `json:"errors"`
 	Items  []struct {
-		Index struct {
-			Status int `json:"status"`
-			Error  struct {
-				Type   string `json:"type"`
-				Reason string `json:"reason"`
-			} `json:"error"`
-		} `json:"index"`
+		Item Item `json:"index"`
```
Comment (iulianpascalau) [inline client/data.go:16]: :+1:

File: client/elasticClientCommon.go [hunk 7]
```diff
@@ -102,32 +102,68 @@ func elasticBulkRequestResponseHandler(res *esapi.Response) error {
 		return fmt.Errorf("%w cannot read elastic response body bytes", err)
 	}
 
-	bulkResponse := &BulkRequestResponse{}
-	err = json.Unmarshal(bodyBytes, bulkResponse)
+	return extractErrorFromBulkBodyResponseBytes(bodyBytes, index)
+}
+
+func extractErrorFromBulkBodyResponseBytes(bodyBytes []byte, index string) error {
+	if index == indexer.TransactionsIndex {
```
Comment (iulianpascalau) [inline client/elasticClientCommon.go:109]: could have used polymorphism instead of this ugly if

File: client/elasticClientCommon.go [hunk 8]
```diff
@@ -102,32 +102,68 @@ func elasticBulkRequestResponseHandler(res *esapi.Response) error {
...
+		return searchForErrorsUpdate(bodyBytes)
+	}
+
+	return searchForErrorsIndex(bodyBytes)
+}
+
+func searchForErrorsUpdate(bodyBytes []byte) error {
+	response := BulkRequestResponseUpdate{}
```
Comment (iulianpascalau) [inline client/elasticClientCommon.go:117]: a lot of duplicated code between searchForErrorsUpdate and searchForErrorsIndex

File: client/elasticClientCommon.go (hunk 4 above)
Comment (miiu96) [reply client/elasticClientCommon.go:116]: done.

File: client/elasticClientCommon.go (hunk 5 above)
Comment (miiu96) [reply client/elasticClientCommon.go:146]: done.

File: client/elasticClientCommon.go (hunk 8 above)
Comment (miiu96) [reply client/elasticClientCommon.go:117]: refactored.

File: client/elasticClientCommon.go (hunk 7 above)
Comment (miiu96) [reply client/elasticClientCommon.go:109]: refactored not need.

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**